frontend/src/metabase/containers/ItemPicker.jsx

```diff
@@ -182,6 +189,7 @@ export default class ItemPicker extends React.Component {
                   ...(models.size === 1
                     ? { model: Array.from(models)[0] }
                     : {}),
+                  ...(writeable ? { writeable: true } : {}),
```

Comment:
If we pass a falsey `writeable` (or don't pass it), it's currently converted to false in the entity. Just `writeable,` should work for this line. The exception is if we might get truthy values here that we want to convert to `true`?

Reply:
haha thanks I feel like this is entry-level javascript cargo culting that I'm doing here, I appreciate these catches. I think we should only allow boolean values

Reply:
what I was going for was to only include a `writeable` key if `writeable` was truthy so that when the API call was made it would only include `writeable` in the query params if it were true

Reply:
Ah ok. In that case, I'd leave this as-is and drop the ` || false` in the [other place](https://github.com/metabase/metabase/pull/10997#discussion_r329044169) I commented.

I'm assuming that passing `{writeable: undefined}` there doesn't actually send the param, but I'm not sure so it's worth checking.

Reply:
Actually, maybe it makes sense to move the `writeable ? { writeable: true } : {}` logic into the entity. Then, my assumption doesn't matter.

Reply:
So have something like this in `metabase/entities/search.js` ?

```javascript
  api: {
    list: async (query = {}) => {
      if (query.collection) {
        const {
          collection,
          archived,
          model,
          writeable,
          ...unsupported
        } = query;
        if (Object.keys(unsupported).length > 0) {
          throw new Error(
            "search with `collection` filter does not support these filters: " +
              Object.keys(unsupported).join(", "),
          );
        }
        return (await collectionList({
          collection,
          archived,
          model,
          ...(writeable ? { writeable: true } : {}) // <--- this?
        })).map(item => ({
          collection_id: canonicalCollectionId(collection),
          archived: archived || false,
          ...item,
        }));
      } else {
        return searchList(query);
      }
    },
  },
```

Reply:
at the same time i guess it's not really a big deal to include `writeable=false` in the query param

Reply:
Yeah that looks good 👍 

Maybe worth checking the behavior of `undefined`. I'd guess that's omitted automatically, but with this code it works either way.